### ISSUE TITLE
Accept null values

### ISSUE THREAD

**Issue by Ragowit** [Ragowit/PsnApiWrapperNet#1]
Some of our model parameters doesn't take null, instead using a default value. This should be fixed so it is null if no response from the API is given.

**Comment by Crashdummyy** [comment]
[NextOffset](https://github.com/Ragowit/PsnApiWrapperNet/blob/master/PsnApiWrapperNet/Model/GameList.cs#L7) can be null for when youre on the last page

**Comment by Ragowit** [comment]
Thanks @Crashdummyy, I'm currently on parental leave but I hope I can get back soon and update some of the code.

**Comment by Crashdummyy** [comment]
Sure Thing ;)
I cant use your library while it uses Restsharp but keep Up the good work

**Comment by Ragowit** [comment]
@Crashdummyy How come? It was a quick and easy way to get http response but I can look into using native .NET stuff.

**Comment by Crashdummyy** [comment]
I am implementing an application using blazor webassembly.
It uses another implementation of the Httpclient and Restsharp does not allow the injection of this.

And [apparently](https://github.com/restsharp/RestSharp/issues/1524)...
[no One](https://github.com/restsharp/RestSharp/issues/1565)
[cares](https://github.com/restsharp/RestSharp/issues/1592)

But using Restsharp is quite comfy, why not keep it that way ;)
Its too late for me anyway 😛 
I just had a peek into your repository to get the RestEndpoints as I cant find any documentation.

**Comment by alexeyzimarev** [comment]
@Crashdummyy that's not correct, your information is based on three old issues that were auto-closed.

```csharp
var client = new RestClient(myHttpClient);
```

works since the v107 is out (about a month now)

**Comment by alexeyzimarev** [comment]
But this projects uses v106, so yes, it won't work, unless it gets an upgrade to the latest RestSharp version.

**Comment by Crashdummyy** [comment]
> @Crashdummyy that's not correct, your information is based on three old issues that were auto-closed.
> 
> ```cs
> var client = new RestClient(myHttpClient);
> ```
> 
> works since the v107 is out (about a month now)

Thats amazing news.
Because I really liked the comfort as well.
However no one cared to comment on one of those issues or reference them in a PR.
Thus I lost interest in using it.

**Comment by alexeyzimarev** [comment]
> However no one cared to comment on one of those issues or reference them in a PR.

No one you mean me? I am the only one maintaining it, with a single sponsor for RestSharp on OpenCollective. I expect people to adjust their expectations accordingly and show some respect.

> Thus I lost interest in using it.

It's ok, I guess. Do what's best for you 🤷‍♂️

**Comment by alexeyzimarev** [comment]
@Ragowit I checked what needs to be done to move to RS 107 for you. The main thing is to make all the functions async. Also, I noticed you keed deserializing responses, whilst RestSharp has deserialization built-in.

Here are a couple of functions that I changed:

```csharp
        private static async Task<string> Authenticate()
        {
            var requestGet = new RestRequest("authz/v3/oauth/authorize")
                .AddParameter("client_id", "ac8d161a-d966-4728-b0ea-ffec22f69edc")
                .AddParameter("redirect_uri", "com.playstation.PlayStationApp://redirect")
                .AddParameter("response_type", "code")
                .AddParameter("scope", "psn:mobile.v1 psn:clientapp");
            var responseGet = await _restClientAuth.GetAsync(requestGet);
            var location = ((string)responseGet.Headers.First(x => x.Name == "Location").Value).Replace("com.playstation.playstationapp://redirect/", "");
            var code = HttpUtility.ParseQueryString(location).Get("code");

            var requestPost = new RestRequest("authz/v3/oauth/token")
                .AddParameter("grant_type", "authorization_code")
                .AddParameter("redirect_uri", "com.playstation.PlayStationApp://redirect")
                .AddParameter("code", code);
            var responsePost = await _restClientAuth.PostAsync<Auth>(requestPost);

            return responsePost.access_token;
        }

        public async Task<GameList> GameList(string accountId, int offset = 0)
        {
            var request = new RestRequest("gamelist/v2/users/{accountId}/titles")
                .AddUrlSegment("accountId", accountId)
                .AddParameter("offset", offset);
            return await _restClientBase.GetAsync<GameList>(request);
        }
```

I'd also say that you might want to delay the authentication call, and maybe wrap the whole auth story to a custom Authorizer. I made an example of it https://gist.github.com/alexeyzimarev/62d77bb25d7aa5bb4b9685461f8aabdd

**Comment by Crashdummyy** [comment]
> > However no one cared to comment on one of those issues or reference them in a PR.
> 
> No one you mean me? I am the only one maintaining it, with a single sponsor for RestSharp on OpenCollective. I expect people to adjust their expectations accordingly and show some respect.
> 
> > Thus I lost interest in using it.
> 
> It's ok, I guess. Do what's best for you man_shrugging

Im sorry if I offended you as I dont meant you on this matter.
I meant just anyone with interest in using your library in blazor wasm.

It could be just me but whenever i want to use a library from github I usually go for the issues to see if I might encounter some Problems

**Comment by Ragowit** [comment]
Woa, thanks a lot @alexeyzimarev !!

I will surely update to the latest RestSharp when I have the time again, I'm still on parental leave for the rest of this month.

**Comment by Ragowit** [comment]
Sorry for ping @Crashdummyy, just wanted to let you know that I recently released 1.0.0 that just uses .NET stuff.
